### PR TITLE
update substrate, wasmi, parity-wasm, wasm-instrument

### PR DESCRIPTION
**Pull request by ukint-vs**
update to this commit https://github.com/gear-tech/substrate/commit/afd2e78fc4a6e7d61497815505329015d7507eef

@gear-tech/dev

### CODE REVIEW

File: node/src/benchmarking.rs [hunk 1]
```diff
@@ -1,6 +1,6 @@
-// This file is part of Gear.
+// This file is part of Substrate.
```
Comment (breathx) [inline node/src/benchmarking.rs:1]: We are `Gear`!

File: node/src/benchmarking.rs [hunk 2]
```diff
@@ -113,10 +169,10 @@ pub fn create_benchmark_extrinsic(
     let signature = raw_payload.using_encoded(|e| sender.sign(e));
 
     runtime::UncheckedExtrinsic::new_signed(
-        call,
+        call.clone(),
         sp_runtime::AccountId32::from(sender.public()).into(),
-        runtime::Signature::Sr25519(signature),
-        extra,
+        runtime::Signature::Sr25519(signature.clone()),
+        extra.clone(),
```
Comment (breathx) [inline node/src/benchmarking.rs]: redundant cloning?

File: pallets/gear/src/benchmarking/code.rs [hunk 3]
```diff
@@ -195,7 +195,7 @@ where
         for func in def.imported_functions {
             let sig = builder::signature()
                 .with_params(func.params)
-                .with_results(func.return_type.into_iter().collect())
+                .with_results(func.return_type.into_iter().collect::<Vec<ValueType>>())
```
Comment (breathx) [inline pallets/gear/src/benchmarking/code.rs:198]:
```suggestion
                .with_results(func.return_type.into_iter().collect::<Vec<_>>())
```

Doesn't it catch it?

**Review by breathx — APPROVED** [review]

**Comment by shamilsan** [comment]
Waiting for `wasmi` update.

https://github.com/paritytech/wasmi/issues/395

**Pull request merged**